File: 2-java-corba/1-hello-world/java/EchoClient.java
import org.omg.CosNaming.*;
import org.omg.CosNaming.NamingContextPackage.*;
import org.omg.CORBA.*;

public class EchoClient {
  public static void main(String args[]) {
    try {
      org.omg.CORBA.Object objRef;
      ORB orb = ORB.init(args, null);
      // create and initialize the ORB
      // get the naming service
      objRef = orb.resolve_initial_references("NameService");
      NamingContextExt ncRef = NamingContextExtHelper.narrow(objRef);
      // resolve the object reference from the naming service
      objRef = ncRef.resolve_str("echo.echo");
      // convert the CORBA object reference into Echo reference
      Echo echoRef = EchoHelper.narrow(objRef);
      // remote method invocation
      String response = echoRef.echoString("coucou");
      System.out.println(response);
    } catch(Exception e) {
      System.out.println("ERROR : " + e);
      e.printStackTrace(System.out);
    }
  }
}

File: 2-java-corba/1-hello-world/c++/EchoClient.cpp
#include <iostream>
#include <omniORB4/CORBA.h>
#include <omniORB4/Naming.hh>
#include "Echo.hh"

using namespace std;

int main (int argc, char **argv) {
  CORBA::Object_var objRef;
  // initialize the ORB
  CORBA::ORB_var orb = CORBA::ORB_init(argc, argv);
  // get the naming service
  objRef = orb->resolve_initial_references("NameService");
  CosNaming::NamingContext_var nsRef =
    CosNaming::NamingContext::_narrow(objRef);
  // resolve the "echo" CORBA object from the naming service
  CosNaming::Name name; name.length(1);
  name[0].id
    = (const char*) "echo";
  name[0].kind = (const char*) "echo";
  objRef = nsRef->resolve(name);
  Echo_var echoRef = Echo::_narrow(objRef);
  // remote method invokation
  cout << echoRef->echoString("coucou") << endl;
  return 0;
}
